contracts/PolygonTokenBridger.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "../Lockable.sol";
+
+interface PolygonIERC20 is IERC20 {
+    function withdraw(uint256 amount) external;
+}
+
+// Because Polygon only allows withdrawals from a particular address to go to that same address on mainnet, we need to
+// have some sort of contract that can guarantee identical addresses on Polygon and Ethereum.
+// Note: this contract is intended to be completely immutable, so it's guaranteed that the contract on each side is
+// configured identically as long as it is created via create2.
+contract PolygonTokenBridger is Lockable {
+    using SafeERC20 for PolygonIERC20;
+    using SafeERC20 for IERC20;
+
+    address public immutable destination;
+
+    constructor(address _destination) {
+        destination = _destination;
+    }
+
+    // Polygon side.
+    function send(PolygonIERC20 token, uint256 amount) public nonReentrant {
+        token.safeTransferFrom(msg.sender, address(this), amount);
+        token.withdraw(amount);
+    }
+
+    // Mainnet side.
+    function retrieve(IERC20 token) public nonReentrant {
```

Comment:
It's annoying that we will need a separate bot to execute this method but we could also call it in the executeRelayerRefund and executeSlowRelay methods which are overridable. This would be similar to how OptimismSpokePool works which also has the related problem of receiving ETH over the bridge but needing to have WETH to send to recipients.

Reply:
The retrieve method would need to be called on the mainnet side, right? executeRelayerRefund/executeSlowRelay are SpokePool methods, which would only happen on polygon, not mainnet right?

Reply:
Good point. I thought this TokenBridger contract is meant to assist with tokens sent from HubPool --> Polygon_SpokePool, and back, so couldn't we use the above strategy to at least automate token retrievals on the Polygon side?

Reply:
I don't think we need any assistance on the polygon side. I thought the tokens get deposited/sent automatically, no?